src/engraving/libmscore/chordlist.cpp

```diff
@@ -1432,6 +1434,799 @@ qreal ChordList::position(const QStringList& names, ChordTokenClass ctc) const
     }
 }
 
+//---------------------------------------------------------
+//   findModifierStartIndices
+//---------------------------------------------------------
+void ParsedChord::findModifierStackIndices()
+{
+    QStringList alterationsList = { "b", "bb", "#", "##", "natural" };
+    QStringList addOmitList = { "omit", "no", "add" };
+    for (int index = 0; index < _tokenList.size(); index++) {
+        const ChordToken& tok = _tokenList.at(index);
+        if (tok.names.size() > 0) {
+            if (alterationsList.contains(tok.names.first())) {
+                alterationStackIndices.push_back(index);
+            } else if (addOmitList.contains(tok.names.first())) {
+                addOmitStackIndices.push_back(index);
+                // To prevent the immediate next accidental to be treated as new modifier
+                if (alterationsList.contains(_tokenList.at(index + 1).names.first())) {
+                    index++;
+                    addOmitStackIndices.push_back(index);
+                }
+            }
+        }
+    }
+    // To know where to end stacking and return cursor to the normal position
+    if (alterationStackIndices.size() > 0) {
+        alterationStackingEnd = alterationStackIndices.last() + 1;
+    }
+    if (addOmitStackIndices.size() > 0) {
+        addOmitStackingEnd = addOmitStackIndices.last() + 1;
+    }
+    // To remove double references to a single modifier like add(b9)
+    // All we need is one index per modifier(the starting index)
+    for (int i = 0; i < addOmitStackIndices.size(); i++) {
+        if (addOmitStackIndices.contains(addOmitStackIndices[i] - 1)) {
+            addOmitStackIndices.removeAt(i);
+        }
+    }
+}
+
+//---------------------------------------------------------
+//   checkQualitySymbolsLetterCase
+//---------------------------------------------------------
+void ParsedChord::checkQualitySymbolsLetterCase(const ChordList* cl)
+{
+    // Changes the letter case of the first letter in quality symbols
+    // Leaves untouched if there is no other replacement
+    if (cl->autoCapitalization) {
```

Comment:
You can simplify the implementation of this method:

if (!cl->autoCapitalization) {
return;
}

...